gym/spaces/dict.py

```diff
@@ -49,28 +50,45 @@ def __init__(self, spaces=None, **spaces_kwargs):
                 space, Space
             ), "Values of the dict should be instances of gym.Space"
         super(Dict, self).__init__(
-            None, None
+            None, None, seed
         )  # None for shape and dtype, since it'll require special handling
 
     def seed(self, seed=None):
-        seed = super().seed(seed)
-        try:
-            subseeds = self.np_random.choice(
-                np.iinfo(int).max,
-                size=len(self.spaces),
-                replace=False,  # unique subseed for each subspace
-            )
-        except ValueError:
-            subseeds = self.np_random.choice(
-                np.iinfo(int).max,
-                size=len(self.spaces),
-                replace=True,  # we get more than INT_MAX subspaces
-            )
-
-        for subspace, subseed in zip(self.spaces.values(), subseeds):
-            seed.append(subspace.seed(int(subseed))[0])
-
-        return seed
+        seeds = []
+        if isinstance(seed, dict):
+            for key, seed_key in zip(self.spaces, seed):
+                assert key == seed_key, print(
+                    "Key value",
+                    seed_key,
+                    "in passed seed dict did not match key value",
+                    key,
+                    "in spaces Dict.",
+                )
+                seeds += self.spaces[key].seed(seed[seed_key])
+        elif isinstance(seed, int):
+            seeds = super().seed(seed)
+            try:
+                subseeds = self.np_random.choice(
+                    np.iinfo(int).max,
+                    size=len(self.spaces),
+                    replace=False,  # unique subseed for each subspace
+                )
+            except ValueError:
+                subseeds = self.np_random.choice(
+                    np.iinfo(int).max,
+                    size=len(self.spaces),
+                    replace=True,  # we get more than INT_MAX subspaces
+                )
+
+            for subspace, subseed in zip(self.spaces.values(), subseeds):
+                seeds.append(subspace.seed(int(subseed))[0])
+        elif seed is None:
+            for space in self.spaces.values():
+                seeds += space.seed(seed)
```

Comment:
Should we only add the main seed of subspace (`subspace.seed()[0]`) to the return value rather than extend all items in `subspace.seed()`?

If the subpace of `Dict` here is another compound space (`Tuple` or `Dict`), the length of the return list could be different `len(d.seed(None)) != len(d.seed(0))`.